testing/trino-testing/src/main/java/io/trino/testing/AbstractTestEngineOnlyQueries.java

```diff
@@ -6049,6 +6049,42 @@ public void testExplainExecuteWithUsing()
         assertEquals(getOnlyElement(result.getOnlyColumnAsSet()), getExplainPlan("SELECT * FROM orders WHERE orderkey < 7", LOGICAL));
     }
 
+    @Test
+    public void testExplainExecuteWithShow()
+    {
+        // case1: The output line is larger than the clipping threshold
+        String largerValuesThresholdQuery = "SHOW FUNCTIONS";
+        String explainOutput = (String) computeActual("EXPLAIN " + largerValuesThresholdQuery).getOnlyValue();
+        String expected1 = getExplainPlan(largerValuesThresholdQuery, DISTRIBUTED);
+        assertThat(explainOutput).isEqualTo(expected1);
+        assertThat(explainOutput).contains("...");
+        assertThat(explainOutput).contains("Row count: ");
+
+        // case2: The output line is smaller than the clipping threshold
+        String smallerValuesThresholdQuery = "" +
+                "SELECT * FROM ( " +
+                "         VALUES" +
+                "             (1, 'a')," +
+                "             (2, 'b')," +
+                "             (3, 'c')" +
+                "    )";
+        String explainOutput2 = (String) computeActual("EXPLAIN " + smallerValuesThresholdQuery).getOnlyValue();
+        String expected2 = getExplainPlan(smallerValuesThresholdQuery, DISTRIBUTED);
+        assertThat(explainOutput2).isEqualTo(expected2);
+        // No truncation
+        assertThat(explainOutput2).doesNotContain("...");
+
+        // case3: ValuesNode with 0 rows.
+        String emptyValuesQuery = "SHOW ROLE GRANTS";
+        String explainOutput3 = (String) computeActual("EXPLAIN " + emptyValuesQuery).getOnlyValue();
+        String expected3 = getExplainPlan(emptyValuesQuery, DISTRIBUTED);
+        assertThat(explainOutput3).isEqualTo(expected3);
+
+        // case4: ValuesNode without output symbols
```

Comment:
You can get Values with no output symbols in a query like `SELECT name FROM nation LEFT JOIN (VALUES 1, 2, 3) ON TRUE;` -- it should have 3 rows but no outputs thanks to pruning.
However, I don't like involving the Optimizer in these tests. Instead, we should have a unit test to see what the `PlanPrinter` produces for a single `PlanNode`.